### PR TITLE
[Merged by Bors] - feat(Data/Set/Basic): Lemmas about pairs

### PR DESCRIPTION
**Pull request by apnelson1**
We add some basic API about two-element sets. This involves moving the 'pair' section further down the file because
of the position of lemmas used in proofs. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — COMMENTED** [review]
This seems good, but I took the opportunity to point out some nits:

**Review by madvorak — APPROVED** [review]
Assuming the few cosmetic issues pointed out by @Ruben-VandeVelde get fixed, I am strongly in favor of merging this PR.

File: Mathlib/Data/Set/Basic.lean [hunk 1]
```diff
@@ -2067,10 +2050,64 @@ theorem mem_diff_singleton_empty {t : Set (Set α)} : s ∈ t \ {∅} ↔ s ∈
...
+  simp [subset_antisymm_iff, insert_subset_iff]; aesop
+#align set.pair_eq_pair_iff Set.pair_eq_pair_iff
+
+theorem pair_diff_left (hne : a ≠ b) : ({a, b} : Set α) \ {a} = {b} := by
+  rw [insert_diff_of_mem _ (mem_singleton a), diff_singleton_eq_self (by simpa)]
+
+theorem pair_diff_right (hne : a ≠ b) : ({a, b} : Set α) \ {b} = {a} := by
+  rw [pair_comm, pair_diff_left hne.symm]
+
+@[simp] theorem pair_subset_iff : {a, b} ⊆ s ↔ a ∈ s ∧ b ∈ s := by
```
Comment (kmill) [inline Mathlib/Data/Set/Basic.lean:2086]: I'm wondering why `Set.insert_subset_iff` isn't a `@[simp]` lemma already. If it were, then this lemma would be proved `by simp`.

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Comment by urkud** [comment]
It looks like `Set.diff_subset` (and possibly other lemmas) now use implicit arguments. Please merge `master`, fix compile, then run `bors merge`. Thanks!
bors d+

**Pull request closed without merging**